src/impl/collection_notifier.cpp

```diff
@@ -286,6 +286,12 @@ void CollectionNotifier::prepare_handover()
     m_sg_version = m_sg->get_version_of_current_transaction();
     do_prepare_handover(*m_sg);
     m_has_run = true;
+
+#ifdef REALM_DEBUG
```

Comment:
Even if assertions are enabled in a release build we don't really want the extra lock acquisitions.